File: 0232-implement-queue-using-stacks/0232-implement-queue-using-stacks.cpp
class MyQueue {
public:
    stack<int> a,b;
    int top;
    MyQueue() {
        
    }
    
    void push(int x) {
        if(a.empty()) top=x;
        a.push(x);
    }
    
    int pop() {
        while(a.size()){
            b.push(a.top());
            a.pop();
        }
        int res = b.top();
        b.pop();
        top=b.size()?b.top():-1;
        while(b.size()){
            a.push(b.top());
            b.pop();
        }
        return res;
    }
    
    int peek() {
        return top;
    }
    
    bool empty() {
        return a.size()==0;
    }
};

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue* obj = new MyQueue();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->peek();
 * bool param_4 = obj->empty();
 */

File: 0232-implement-queue-using-stacks/0232-implement-queue-using-stacks.py
from copy import deepcopy

class MyQueue:

    def __init__(self):
        self.stack = []

    def push(self, x: int) -> None:
        st2 = []
        while self.stack:
            st2.append(self.stack.pop())
        
        self.stack.append(x)
        while st2:
            self.stack.append(st2.pop())

    def pop(self) -> int:
        return self.stack.pop()

    def peek(self) -> int:
        return self.stack[-1]

    def empty(self) -> bool:
        return len(self.stack)==0


# Your MyQueue object will be instantiated and called as such:
# obj = MyQueue()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.peek()
# param_4 = obj.empty()